mirror.js

```diff
@@ -19,16 +19,19 @@ async function handleRequest(event) {
   const cacheKey = new Request(cacheUrl.toString(), request);
 
   let response = await cache.match(cacheKey);
-  const imageRequest = new Request(imageURL, {
-    headers: request.headers
-  });
-
+  
   if (!response) {
+    const imageRequest = new Request(imageURL, {
+      headers: request.headers
+    });
     response = await fetch(imageRequest, options);
     response = new Response(response.body, response);
     response.headers.append('Cache-Control', 's-maxage=604800');
     event.waitUntil(cache.put(cacheKey, response.clone()));
+  } else {
+    response = new Response(response.body, response);
```

Comment:
If you don't add this `response.headers.set` is not working?

Reply:
yes correct :P it needs to be in `Response it seems`.. else getting `500 error`..  to debug the error cache doesnt work in local :P so better to have to this like in if condition :)